Sources/SmithyFormURL/Writer.swift

```diff
@@ -0,0 +1,207 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import protocol SmithyReadWrite.SmithyWriter
+import enum SmithyReadWrite.Document
+import enum SmithyTimestamps.TimestampFormat
+import struct SmithyTimestamps.TimestampFormatter
+import struct Foundation.Data
+import struct Foundation.Date
+import struct Foundation.CharacterSet
+
+public final class Writer: SmithyWriter {
```

Comment:
This type is a Writer that will output FormURL data once the tree of writers has been constructed by serialization code.

It effectively replaces the FormURLEncoder that was previously in ClientRuntime.